### ISSUE TITLE
Missing `setGenerationTimeMs` after upgrade to 4.x

### ISSUE THREAD

**Issue by HollyPony** [piwik/piwik#16853]
According to the changelog there is no mentions of this breaking change that was required following documentation of 3.x .

I can see there was some "optimization" undocumented (mainly deleted doc).

Could I have more informations about why it was deleted and how to safely do elsething ?

**Comment by tsteur** [comment]
This seems to be missing from the changelog indeed. Thanks for mentioning it @HollyPony 

To fix this issue we need to mention the removal in the developer changelog.

It was removed because the generation time feature was replaced by the new performance feature in Matomo 4. You can see the more granular/detailed performance metrics now in `Behaviour -> Performance` report. There is currently no replacement. Do you maybe have a single page application or what do you need the method for?

**Comment by HollyPony** [comment]
> To fix this issue we need to mention the removal in the developer changelog.

So, the removal is https://github.com/matomo-org/matomo/pull/15736/commits/f8dfb506dd4ea848af4de5ad06083832b10196e7 from PR https://github.com/matomo-org/matomo/pull/15736

> Do you maybe have a single page application or what do you need the method for?

Yes, I'm maintaining an old angularjs 1 app so ... :shrug: and to be honest I just followed the doc so I'm not sure if it's really relevant to continue using `setGenerationTimeMs` in my case. I really don't know I'm just surprise. 

Nb. I noticed that due to my usage as `tracker.setGenerationTimeMs()` nor the `paq.push("blabla")` then my console crashed.

The fact is, I just don't know if it's simply "ok" to remove that or if I should investigate new purposes about SPAs integrations
